api/agent_client.go

```diff
@@ -249,6 +249,18 @@ type AgentJobState struct {
 
 // GetJobState gets the state of a specific job.
 func (c *AgentClient) GetJobState(ctx context.Context, id string) (result *AgentJobState, retryAfter time.Duration, err error) {
+	// Use the batch API as a shim when useStackAPI is on.
+	if c.useStackAPI {
```

Comment:
i think if we have the reader method (`c.UseStackAPI()`) we should probably use it even internally